media-sound/mixxx/mixxx-9999.ebuild

```diff
@@ -14,119 +14,156 @@ if [[ ${PV} == *9999 ]] ; then
-	opus? (	media-libs/opusfile )
-	qtkeychain? ( dev-libs/qtkeychain:=[qt5(+)] )
-	shout? ( >=media-libs/libshout-2.4.5 )
+	midi? ( media-libs/portmidi )
+	modplug? ( media-libs/libmodplug )
+	mp3? (
+		media-libs/libid3tag:=
+		media-libs/libmad
+	)
+	mp4? ( media-libs/libmp4v2 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	qtkeychain? ( >=dev-libs/qtkeychain-0.14.2:=[qt6(+)] )
+	rubberband? ( media-libs/rubberband:= )
+	shout? ( dev-libs/openssl:= )
+	upower? (
+		dev-libs/glib:2
+		sys-power/upower:=
+	)
 	wavpack? ( media-sound/wavpack )
 "
 DEPEND="${RDEPEND}
 	dev-cpp/ms-gsl
+	test? ( dev-cpp/gtest )
 "
 BDEPEND="
-	dev-qt/qttest:5
-	dev-qt/qtxmlpatterns:5
+	dev-util/spirv-tools
 	virtual/pkgconfig
 "
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-9999-docs.patch
+	# Fix strict-aliasing violations in vendored katai_cpp_stl_runtime
+	# https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/commit/c01f530.patch
+	"${FILESDIR}"/${PN}-2.5.0-fix-strict-aliasing-kaitai.patch
+)
+
+CMAKE_SKIP_TESTS=(
+	# need HID controller
+	LegacyControllerMappingValidationTest.HidMappingsValid
+	# randomly fails
+	# https://github.com/mixxxdj/mixxx/issues/12554
+	EngineBufferE2ETest
 )
 
 src_configure() {
+	# prevent ld error as package builds static libs.
+	tc-is-lto && append-flags $(test-flags -ffat-lto-objects)
+
 	local mycmakeargs=(
-		-DFAAD="$(usex aac on off)"
-		-DFFMPEG="$(usex ffmpeg on off)"
-		-DHID="$(usex hid on off)"
-		-DLILV="$(usex lv2 on off)"
-		-DMAD="$(usex mp3 on off)"
-		-DOPTIMIZE="off"
-		-DCCACHE_SUPPORT="off"
-		-DOPUS="$(usex opus on off)"
-		-DBROADCAST="$(usex shout on off)"
-		-DVINYLCONTROL="on"
+		-DBATTERY="$(usex upower)"
+		-DBROADCAST="$(usex shout)"
+		-DBUILD_SHARED_LIBS=ON
+		-DBUILD_TESTING="$(usex test)"
+		-DBUILD_BENCH="$(usex benchmark)"
+		# prevent duplicate call
+		-DCCACHE_SUPPORT=OFF
+		-DENGINEPRIME=OFF
+		-DFAAD="$(usex aac)"
+		-DFFMPEG="$(usex ffmpeg)"
+		-DGPERFTOOLS="$(usex benchmark)"
+		-DGPERFTOOLSPROFILER="$(usex benchmark)"
+		-DHID=ON
 		-DINSTALL_USER_UDEV_RULES=OFF
-		-DWAVPACK="$(usex wavpack on off)"
-		-DQTKEYCHAIN="$(usex qtkeychain on off)"
-		-DKEYFINDER="$(usex keyfinder on off)"
-		-DDOWNLOAD_MANUAL=OFF
-		-DBUILD_SHARED_LIBS=OFF
+		-DKEYFINDER="$(usex keyfinder)"
+		-DLILV="$(usex lv2)"
+		-DMAD="$(usex mp3)"
+		-DMODPLUG="$(usex modplug)"
+		-DOPTIMIZE=OFF
+		-DOPUS="$(usex opus)"
+		-DPORTMIDI="$(usex midi)"
+		-DQML="$(usex qml)"
+		-DQTKEYCHAIN="$(usex qtkeychain)"
+		-DRUBBERBAND="$(usex rubberband)"
+		-DVINYLCONTROL=ON
+		-DWAVPACK="$(usex wavpack)"
 	)
 
-	if [[ ${PV} == 9999 ]] ; then
-		mycmakeargs+=(
-			-DENGINEPRIME="OFF"
-		)
-	fi
 	cmake_src_configure
 }
 
 src_install() {
 	cmake_src_install
+	use benchmark && dobin "${BUILD_DIR}"/mixxx-test
 	udev_newrules "${S}"/res/linux/mixxx-usb-uaccess.rules 69-mixxx-usb-uaccess.rules
 
-	if use doc ; then
-		dodoc README res/Mixxx-Keyboard-Shortcuts.pdf
-	fi
+	dodoc res/Mixxx-Keyboard-Shortcuts.pdf
```

Comment:
What happened to README? I can see there is `README.md` and it gets installed by their CMake build system, together with Mixxx-Keyboard-Shortcuts.pdf - are we okay with that, is it in the right path, and if so, is this dodoc simply superfluous?

Reply:
dodoc is indeed superfluous